GiniVision/Classes/Core/GiniPDFDocument.swift

```diff
@@ -64,7 +64,7 @@ final public class GiniPDFDocument: NSObject, GiniVisionDocument {
     
     public func checkType() throws {
         if self.data.isPDF {
-            if case 1...MAX_PDF_PAGES_COUNT = self.numberPages {
+            if case 1...GiniPDFDocument.maxPagesCount = self.numberPages {
```

Comment:
Same thing. If Why would a Document know about the 10 page constraint. a 1000 page PDF is still a valid PDF document. This constraint comes later, in another business logic layer.

Reply:
It is part of their own validation check.

Reply:
What does that mean? Whose's validation check?

Reply:
Each document has different kinds of validation.